algebird-test/src/main/scala/com/twitter/algebird/ApproximateProperty.scala

```diff
@@ -0,0 +1,137 @@
+package com.twitter.algebird
+
+import org.scalacheck.{ Gen, Prop }
+import org.scalacheck.util.Pretty
+
+trait ApproximateProperty {
+  type Exact
+  type Approx
+  type Input
+  type Result
+
+  def exactGenerator: Gen[Exact]
+  def inputGenerator(e: Exact): Gen[Input]
+
+  def makeApproximate(e: Exact): Approx
+
+  def exactResult(e: Exact, i: Input): Result
+  def approximateResult(a: Approx, i: Input): GeneralizedApproximate[Result]
+}
+
+trait GeneralizedApproximate[T] {
+  def withProb: Double
+  def contains(t: T): Boolean
```

Comment:
Actually, what you are trying to do here is add a common interface to ApproximateBoolean and `Approximate[N]` right?

Maybe just do that directly rather than through implicit conversion.